hkmc2/shared/src/main/scala/hkmc2/syntax/Keyword.scala

```diff
@@ -63,7 +63,11 @@ object Keyword:
   
   val `if` = Keyword("if", N, nextPrec)
   val `while` = Keyword("while", N, curPrec)
-  val `then` = Keyword("then", nextPrec, curPrec)
+  
+  val thenPrec = nextPrec
+  val `then` = Keyword("then", thenPrec, curPrec)
+  val `do` = Keyword("do", thenPrec, curPrec)
```

Comment:
It seems this is equivalent but clearer?

```suggestion
  val thenPrec = nextPrec
  val `then` = Keyword("then", thenPrec, thenPrec)
  val `do` = Keyword("do", thenPrec, thenPrec)
```